application/src/main/kotlin/com/studentcenter/weave/application/user/service/util/impl/strategy/AppleOpenIdTokenResolveStrategy.kt

```diff
@@ -1,13 +1,34 @@
 package com.studentcenter.weave.application.user.service.util.impl.strategy
 
+import com.studentcenter.weave.application.common.properties.OpenIdProperties
 import com.studentcenter.weave.application.user.vo.UserTokenClaims
+import com.studentcenter.weave.domain.user.enums.SocialLoginProvider
+import com.studentcenter.weave.domain.user.vo.Nickname
+import com.studentcenter.weave.support.common.vo.Email
+import com.studentcenter.weave.support.security.jwt.util.JwtTokenProvider
+import com.studentcenter.weave.support.security.jwt.vo.JwtClaims
 import org.springframework.stereotype.Component
+import java.net.URL
 
 @Component
-class AppleOpenIdTokenResolveStrategy : OpenIdTokenResolveStrategy {
+class AppleOpenIdTokenResolveStrategy(
+    private val openIdProperties: OpenIdProperties
+) : OpenIdTokenResolveStrategy {
 
-    override fun resolveIdToken(idToken: String): UserTokenClaims.IdToken {
-        TODO("Not yet implemented")
+    companion object {
+        const val DEFAULT_NICKNAME = ""
     }
 
+    override fun resolveIdToken(idToken: String): UserTokenClaims.IdToken {
+        val jwksUri =
+            URL(this.openIdProperties.providers.getValue(SocialLoginProvider.APPLE).jwksUri)
```

Comment:
p3) jwksUri는 동적으로 항상 조회해야 하는 이유가 있나요?
```suggestion
        val jwksUri =
            URL(this.openIdProperties.providers.getValue(SocialLoginProvider.APPLE).jwksUri)
```

Reply:
생각해 보니 어차피 URI가 변할 일은 없겠군요

Reply:
> 생각해 보니 어차피 URI가 변할 일은 없겠군요

어떻게 처리하실 계획이신가요? 🤔

Reply:
> 어떻게 처리하실 계획이신가요? 🤔

각 클래스 내에 상수로 관리하는 것이 적합하지 않을까요.!!

Reply:
provider측에서 jwks uri 를 변경하면 코드도 변경될것같아서 별도 properties 로 변경했어요, 저는 이게 db connection uri랑 다를바 없다생각해요